core/txindexer.go

```diff
@@ -159,15 +155,19 @@ func (indexer *txIndexer) loop(chain *BlockChain) {
 				stop = make(chan struct{})
 				done = make(chan struct{})
 				indexer.chain.wg.Add(1)
-				go indexer.run(rawdb.ReadTxIndexTail(indexer.db), head.Block.NumberU64(), stop, done)
+				go func() {
```

Comment:
If this must always be the same as the equivalent go routine then consider abstracting it into shared code, otherwise all LGTM.